src/scraper.rs

```diff
@@ -182,16 +183,27 @@ impl Scraper {
 
         dom.find_urls_as_strings()
             .into_iter()
-            .filter(|candidate| Scraper::should_visit(candidate, &url))
+            .filter(|candidate| Scraper::should_visit(candidate))
             .for_each(|next_url| {
-                let mut next_full_url = url.join(&next_url).unwrap();
+                let mut url_to_parse = next_url.clone();
+                // If url is `///tools.wmflabs.org/`, the app crashes
```

Comment:
This should be a `FIXME` comment rather than just a comment, and maybe we'll have to open an issue associated with that which we could reference here via its number.

Or are you saying that any URL with leading three slashes crashes the app? Which is why you remove it on the next line?

In that case, I think it would be better to add a generic comment, such as `// If the URL starts with three slashes, the app crashes. Replace them by two slashes`

Reply:
Fixed in the latest commit. Thanks!